### DISCUSSION TITLE
Is it appropriate to use typescript-written next-i18next config object?

### DISCUSSION THREAD

**Discussion by knapcp3** [i18next/next-i18next#2293] [Q&A] [answered]
I would love to have the option to write next-i18next in typescript to be able to import other `ts` files there (otherwise I have to use `cjs` files, for example for defining default translations). I removed `next-i18next.config.js` file and defined the config file in `nextI18NextConfig.ts` file as 
```
import {defaultTranslation} from 'some-ts-file';
import resourcesToBackend from 'i18next-resources-to-backend';

import ChainedBackend from 'i18next-chained-backend';
import HttpBackend from 'i18next-http-backend';
import i18nConfig from './i18n';

const bundledResources = {
  en: defaultTranslation,
};

const nextI18NextConfig = {
  debug: false,
  i18n: i18nConfig,
  ns: ['common'],
  use: [ChainedBackend],
  backend: {
    // eslint-disable-next-line @typescript-eslint/no-unsafe-call
    backends: [resourcesToBackend(bundledResources), HttpBackend],
    backendOptions: [
      {},
      {
        loadPath: `...`,
      },
    ],
  },

  serializeConfig: false,
};

export default nextI18NextConfig;
```

and then imported this config object into `serverSideTranslations` and `appWithTranslation` as presented in https://github.com/i18next/next-i18next?tab=readme-ov-file#unserializable-configs. It works but I'm wondering if this is more of a hacky solution or something that the author intentionally gave the option to achieve. @adrai could you share if you think this a valid solution?

**Comment by adrai** [comment, marked as answer]
If it works then it's valid 😉
